code/go/internal/validator/semantic/validate_kibana_matching_object_ids.go

```diff
@@ -47,8 +47,9 @@ func ValidateKibanaObjectIDs(fsys fspath.FS) ve.ValidationErrors {
 				continue
 			}
 
-			if ruleID != objectID {
-				errs = append(errs, errors.New("rule ID is different from the object ID"))
+			if !strings.HasPrefix(objectID.(string), ruleID.(string)) {
```

Comment:
These type assertions could result in a panic if `objectId` or `ruleID` are not string.
Could you add those checks before this if statement ? Something like:
```suggestion
            stringObjectID, ok := objectID.(string)
            if !ok {
                errs = append(errs, errors.Wrapf(err, ...)
                continue
            }
            stringRuleID, ok := ruleID.(string)
            if !ok {
                errs = append(errs, errors.Wrapf(err, ...)
                continue
            }
			if !strings.HasPrefix(stringObjectID, stringRuleID) {
```

Reply:
@mrodm Thanks for the suggestion 👍 Added additional checks